include/flyft/grid_interpolator.h

```diff
@@ -0,0 +1,32 @@
+#ifndef FLYFT_GRID_INTERPOLATOR_H_
+#define FLYFT_GRID_INTERPOLATOR_H_
+
+#include "flyft/three_dimension_index.h"
+#include <iostream>
+#include <tuple>
+#include <cmath>
+
+namespace flyft
+{
+
+class GridInterpolator
+    {  
+    public:
+        GridInterpolator(std::string s);
+        GridInterpolator(std::string s, int ni, int nj, int nk);
+        GridInterpolator(std::string s,int ni, int nj, int nk, double dx, double dy, double dz);
```

Comment:
I don't think you need all of these constructors since we are just hacking this together. If the file is going to have the bounds in it as the first 6 values, then do like this:
```suggestion
        GridInterpolator(const std::string& s);
```
FYI: We use the const and & for strings to make sure it is passed by const reference.

Reply:
I have tried to implement it without the constructors, but it gives me this error:
`no default constructor exists for class "flyft::ThreeDimensionIndex"C/C++(291)`, even though I give the constructor for the ThreeDimensionIndex later within the constructor.

Reply:
You need to implement a default constructor (the constructor that takes no arguments) for the ThreeDimensionIndex then! Fill the dimensions and spacings with zeros, for example

Reply:
Okay!